pkg/api/vlabs/validate_test.go

```diff
@@ -1439,6 +1436,80 @@ func TestProperties_ValidateAddon(t *testing.T) {
 		t.Errorf("expected error with message : %s, but got : %s", expectedMsg, err.Error())
 	}
 }
+func TestProperties_ValidateZones(t *testing.T) {
```

Comment:
It looks like you're running multiple testcases and modifying AgentPoolProperties in within one Go t execution. Let's have multiple go subtests